internal/services/extendedlocation/extended_location_custom_location_test.go

```diff
@@ -0,0 +1,351 @@
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data, credential, privateKey, publicKey),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccExtendedLocationCustomLocations_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_extended_custom_location", "test")
+	r := CustomLocationResource{}
+	credential, privateKey, publicKey := r.getCredentials(t)
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data, credential, privateKey, publicKey),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.update(data, credential, privateKey, publicKey),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.basic(data, credential, privateKey, publicKey),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func (r CustomLocationResource) basic(data acceptance.TestData, credential string, privateKey string, publicKey string) string {
+	template := r.template(data, credential, publicKey, privateKey)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_extended_custom_location" "test" {
+  name                = "acctestcustomlocation%[2]d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  cluster_extension_ids = [
+    azurerm_arc_kubernetes_cluster_extension.test.id
+  ]
+  namespace        = "namespace%[2]d"
+  host_resource_id = azurerm_arc_kubernetes_cluster.test.id
+}
+`, template, data.RandomInteger)
+}
+
+func (r CustomLocationResource) update(data acceptance.TestData, credential string, privateKey string, publicKey string) string {
+	template := r.template(data, credential, publicKey, privateKey)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_extended_custom_location" "test" {
+  name                = "acctestcustomlocation%[2]d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  cluster_extension_ids = [
+    azurerm_arc_kubernetes_cluster_extension.test.id
+  ]
+
+  display_name     = "customlocationupdate%[2]d"
+  namespace        = "namespace%[2]d"
+  host_resource_id = azurerm_arc_kubernetes_cluster.test.id
+}
+`, template, data.RandomInteger)
+}
+
+func (r CustomLocationResource) template(data acceptance.TestData, credential string, publicKey string, privateKey string) string {
+	data.Locations.Primary = "eastus"
+	provisionTemplate := r.provisionTemplate(data, credential, privateKey)
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {
+    resource_group {
+      prevent_deletion_if_contains_resources = false
```

Comment:
Using this feature will potentially mask destroy issues, can we get this working without this feature flag?